showcases/src/captioned-box.ts

```diff
@@ -13,7 +13,13 @@ export const renderCaptionedBox = ({
   showcaseStyle,
   captionWidth,
   componentClass,
-  hasCheckeredBackground,
+  hasCheckeredBackground = false,
+}: {
+  showcaseClass?: string;
+  showcaseStyle?: string;
+  captionWidth?: string;
+  componentClass?: string;
+  hasCheckeredBackground: boolean;
```

Comment:
as pointed out in https://github.com/divriots/dockit-core/pull/86/files#r863616130, the proper way is to call these methods separately
IMO the main issue is that you are call 2 different methods in the same place and try to pass different set of arguments to them, incompatible, but a little bit overlapping
you can workaround this by making all arguments optional, but this is not good either
please consider calling them like

```
      ${
        type === 'box'
          ? renderCaptionedBox({
              [showcaseAttr]: showcase,
              captionWidth,
              componentClass,
              hasCheckeredBackground,
            })
          : renderCaptionedText({
            [showcaseAttr]: showcase,
              captionWidth,
              hasLongText,
            })
      }
```

and revisit types of these methods by making correct optional and required ones

also check this out https://stackoverflow.com/a/60617060
this is a pretty cool way to say 1 of 2 is required, but not both at the same time

Reply:
I'll address this refactoring in another change, as it's not the purpose of this one.